### PR TITLE
New serverless pattern- textract-lambda-cdk-dotnet

### PR DESCRIPTION
**Pull request by PankajRawat333**
*Issue #, if available:*

*Description of changes:*
This CDK application demonstrates how to extract text from image using Amazon Textract, AWS Lambda function and Amazon S3.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: textract-lambda-cdk-dotnet/Cdk/README.md
Comment (bfreiberg) [inline textract-lambda-cdk-dotnet/Cdk/README.md:1]: can this be removed?

Comment (bfreiberg) [inline textract-lambda-cdk-dotnet/Cdk/README.md:1]: Please also rename the folder to either `cdk` to `CDK`

File: textract-lambda-cdk-dotnet/Cdk/src/Cdk/CdkStack.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+using Constructs;
+
+namespace Cdk
+{
+    public class CdkStack : Stack
+    {
+        readonly string lambdaFunctionName = "textractLambdaFunction";
+        internal CdkStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
+        {
+            //Create S3 bucket
```
Comment (bfreiberg) [inline textract-lambda-cdk-dotnet/Cdk/src/Cdk/CdkStack.cs:16]: I appreciate the comment but it doesn't add any value in my opinion. I suggest removing them and adding comments wherever something might not be immediately obvious

File: textract-lambda-cdk-dotnet/Cdk/src/Cdk/CdkStack.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+            //Lambda Function IAM role
+            var lambdaIAMRole = new Role(this, "TextractLambdaRole", new RoleProps
+            {
+                AssumedBy = new ServicePrincipal("lambda.amazonaws.com"),
+                Description = "IAM Role for the lambda function",
+                ManagedPolicies = new[]
+                {
+                        ManagedPolicy.FromAwsManagedPolicyName("service-role/AWSLambdaBasicExecutionRole"),
+                        ManagedPolicy.FromAwsManagedPolicyName("AmazonTextractFullAccess")
```
Comment (bfreiberg) [inline textract-lambda-cdk-dotnet/Cdk/src/Cdk/CdkStack.cs:30]: Is `FullAccess` really necessary here? Please try to apply the principle of least privilege wherever reasonably possible

File: textract-lambda-cdk-dotnet/Cdk/src/Cdk/CdkStack.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            var bucket = new Bucket(this, "textract-bucket", new BucketProps
+            {
+                BucketName = $"{this.Account}-textract-bucket"
+            });
+
+            //Lambda Function IAM role
+            var lambdaIAMRole = new Role(this, "TextractLambdaRole", new RoleProps
+            {
+                AssumedBy = new ServicePrincipal("lambda.amazonaws.com"),
+                Description = "IAM Role for the lambda function",
```
Comment (bfreiberg) [inline textract-lambda-cdk-dotnet/Cdk/src/Cdk/CdkStack.cs:26]:
```suggestion
                Description = "IAM Role for the Lambda function",
```

File: textract-lambda-cdk-dotnet/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# Extract text from image using Amazon Textract
+
+This CDK application demonstrates how to extract text from image using Amazon Textract, AWS Lambda function and Amazon S3.
```
Comment (bfreiberg) [inline textract-lambda-cdk-dotnet/README.md:3]:
```suggestion
This CDK application demonstrates how to extract text from images using Amazon Textract, AWS Lambda function and Amazon S3.
```

File: textract-lambda-cdk-dotnet/README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+After deployment, add an image to the `input/` folder of Amazon S3 bucket. After few seconds, check the `output/` folder of S3 bucket for extracted text.
+
+## Cleanup
+Run the given command to delete the resources that were created. It might take some time for the CloudFormation stack to get deleted.
+```
+cdk destroy
+```
+
+----
+Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (bfreiberg) [inline textract-lambda-cdk-dotnet/README.md:52]:
```suggestion
Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

File: textract-lambda-cdk-dotnet/example-pattern.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{
+    "title": "Extract text from image using Amazon Textract",
```
Comment (bfreiberg) [inline textract-lambda-cdk-dotnet/example-pattern.json:2]:
```suggestion
    "title": "Extract text from images using Amazon Textract",
```

File: textract-lambda-cdk-dotnet/example-pattern.json [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    "description": "Create an AWS Lambda function that receives DynamoDB change items.",
```
Comment (bfreiberg) [inline textract-lambda-cdk-dotnet/example-pattern.json:3]: This needs to adjusted

File: textract-lambda-cdk-dotnet/example-pattern.json
Comment (bfreiberg) [inline textract-lambda-cdk-dotnet/example-pattern.json:1]: Please include relevant links under the `"resources"` key, such as object notifications or Textract documentation

File: textract-lambda-cdk-dotnet/Cdk/README.md
Comment (PankajRawat333) [reply textract-lambda-cdk-dotnet/Cdk/README.md:1]: Done

File: textract-lambda-cdk-dotnet/Cdk/src/Cdk/CdkStack.cs (hunk 1 above)
Comment (PankajRawat333) [reply textract-lambda-cdk-dotnet/Cdk/src/Cdk/CdkStack.cs]: Done. Remove comments

File: textract-lambda-cdk-dotnet/Cdk/src/Cdk/CdkStack.cs (hunk 2 above)
Comment (PankajRawat333) [reply textract-lambda-cdk-dotnet/Cdk/src/Cdk/CdkStack.cs]: Good catch! Done

File: textract-lambda-cdk-dotnet/Cdk/src/Cdk/CdkStack.cs (hunk 3 above)
Comment (PankajRawat333) [reply textract-lambda-cdk-dotnet/Cdk/src/Cdk/CdkStack.cs]: Done

File: textract-lambda-cdk-dotnet/README.md (hunk 4 above)
Comment (PankajRawat333) [reply textract-lambda-cdk-dotnet/README.md]: Done

File: textract-lambda-cdk-dotnet/README.md (hunk 5 above)
Comment (PankajRawat333) [reply textract-lambda-cdk-dotnet/README.md]: Done

File: textract-lambda-cdk-dotnet/example-pattern.json (hunk 6 above)
Comment (PankajRawat333) [reply textract-lambda-cdk-dotnet/example-pattern.json]: Done

File: textract-lambda-cdk-dotnet/example-pattern.json (hunk 7 above)
Comment (PankajRawat333) [reply textract-lambda-cdk-dotnet/example-pattern.json]: Done

File: textract-lambda-cdk-dotnet/example-pattern.json
Comment (PankajRawat333) [reply textract-lambda-cdk-dotnet/example-pattern.json:1]: Done

**Comment by bfreiberg** [comment]
Thanks for your contribution @PankajRawat333 . A Developer Advocate will merge the changes and publish the pattern to [serverlessland.com](http://serverlessland.com/).

**Comment by PankajRawat333** [comment]
Hi @bfreiberg may I know the reason of adding extra file (textract-lambda-cdk-dotnet/textract-lambda-cdk-dotnet.json)? It was not the part of [instruction](https://github.com/aws-samples/serverless-patterns/issues/new?assignees=jbesw&labels=&template=new-serverless-pattern-submission.md&title=New+pattern+submission) Should we include this JSON file in future pull requests?

https://github.com/aws-samples/serverless-patterns/issues/new?assignees=jbesw&labels=&template=new-serverless-pattern-submission.md&title=New+pattern+submission

**Pull request merged**